sklearn/metrics/_plot/precision_recall_curve.py

```diff
@@ -540,3 +634,228 @@ def from_predictions(
+            )
+
+        train_size, test_size = (
+            len(cv_results["indices"]["train"][0]),
+            len(cv_results["indices"]["test"][0]),
+        )
+
+        if _num_samples(X) != train_size + test_size:
+            raise ValueError(
+                "X does not contain the correct number of samples. "
+                f"Expected {train_size + test_size}, got {_num_samples(X)}."
+            )
+
+        if fold_name is None:
+            # create iterable of same length as number of precision-recall curves
+            fold_name_ = [None] * len(cv_results["estimator"])
+        elif fold_name is not None and len(fold_name) != len(cv_results["estimator"]):
+            raise ValueError(
+                "When `fold_name` is provided, it must have the same length as "
+                "the number of precision-recall curves to be plotted. Got "
+                f"{len(fold_name)} names instead of {len(cv_results['estimator'])}."
+            )
+        else:
+            fold_name_ = fold_name
+
+        if fold_line_kw is None:
+            fold_line_kw = [
+                {"alpha": 0.5, "color": "tab:blue", "linestyle": "--"}
+            ] * len(cv_results["estimator"])
+        elif isinstance(fold_line_kw, Mapping):
+            fold_line_kw = [fold_line_kw] * len(cv_results["estimator"])
+        elif len(fold_line_kw) != len(cv_results["estimator"]):
+            raise ValueError(
+                "When `fold_line_kw` is a list, it must have the same length as "
+                "the number of precision-recall curves to be plotted."
+            )
+
+        precision_all = []
+        recall_all = []
+        ap_all = []
+        for estimator, test_indices, name in zip(
+            cv_results["estimator"], cv_results["indices"]["test"], fold_name_
+        ):
+            y_true = _safe_indexing(y, test_indices)
+            y_pred, pos_label_ = _get_response_values_binary(
+                estimator,
+                _safe_indexing(X, test_indices),
+                response_method=response_method,
+                pos_label=pos_label,
+            )
+            # Should we use `_validate_from_predictions_params` here?
+            # The check would technically only be needed once though
+            precision, recall, _ = precision_recall_curve(
+                y_true,
+                y_pred,
+                pos_label=pos_label,
+                sample_weight=sample_weight,
+                drop_intermediate=drop_intermediate,
+            )
+            # Note `pos_label` cannot be `None` (default=1), unlike other metrics
+            # such as roc_auc
+            average_precision = average_precision_score(
+                y_true, y_pred, pos_label=pos_label_, sample_weight=sample_weight
+            )
+            # Append all
+            precision_all.append(precision)
+            recall_all.append(recall)
+            ap_all.append(average_precision)
```

Comment:
Don't like this but not sure on the zip suggested in https://github.com/scikit-learn/scikit-learn/pull/30399#discussion_r1880454815 as you've got to unpack at the end :thinking: